aptible/resource_environment_test.go

```diff
@@ -0,0 +1,138 @@
+package aptible
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/aptible/go-deploy/aptible"
+)
+
+func TestAccResourceEnvironment_validation(t *testing.T) {
+	requiredAttrs := []string{"handle", "stack_id", "org_id"}
+	var testSteps []resource.TestStep
+
+	for _, attr := range requiredAttrs {
+		testSteps = append(testSteps, resource.TestStep{
+			PlanOnly:    true,
+			Config:      `resource "aptible_environment" "test" {}`,
+			ExpectError: regexp.MustCompile(fmt.Sprintf("%q is required", attr)),
+		})
+	}
+
+	testSteps = append(testSteps, resource.TestStep{
+		PlanOnly: true,
+		Config: fmt.Sprintf(`
+				resource "aptible_environment" "test" {
+					handle = "%s"
+					org_id = "%s"
+					stack_id = "%v"
+				}
+			`, "test", "invalid-uuid", testStackId),
+		ExpectError: regexp.MustCompile(fmt.Sprintf(`expected %q to be a valid UUID, got invalid-uuid`, "org_id")),
+	})
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckEnvironmentDestroy,
+		Steps:             testSteps,
+	})
+}
+
+func testAccCheckEnvironmentDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*aptible.Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aptible_environment" {
+			continue
+		}
+
+		envID, err := strconv.Atoi(rs.Primary.Attributes["env_id"])
+		if err != nil {
+			return err
+		}
+
+		metricDrain, err := client.GetEnvironment(int64(envID))
+		log.Println("Deleted? ", metricDrain.Deleted)
+		if !metricDrain.Deleted {
+			return fmt.Errorf("environment %v not removed", envID)
+		}
```

Comment:
Optional: Copy pasta

```suggestion
		environment, err := client.GetEnvironment(int64(envID))
		log.Println("Deleted? ", environment.Deleted)
		if !environment.Deleted {
			return fmt.Errorf("environment %v not removed", envID)
		}
```